### DISCUSSION TITLE
javaee-api

### DISCUSSION THREAD

**Discussion by heatherita** [flowlogix/flowlogix#314] [Q&A] [answered]
Should I include javaee-api as a dependency if I am using flowlogix-bom and `<DependencyManagement>` ?

If I use this:
 ```
<dependencies>
   <dependency>
            <groupId>javax</groupId>
            <artifactId>javaee-api</artifactId>
            <type>jar</type>
        </dependency>        
    </dependencies>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.flowlogix</groupId>
                <artifactId>flowlogix-bom</artifactId>
                <version>4.0.7</version>            
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement> 

```
I get this error:

`error reading /home/hpropes/.m2/repository/javax/javaee-api/8.0.1/javaee-api-8.0.1.jar; zip END header not found`

and the jar file is empty. 

Yet if I delete /home/hpropes/.m2/repository/javax/javaee-api and remove: 

```
<dependency>
            <groupId>javax</groupId>
            <artifactId>javaee-api</artifactId>
            <type>jar</type>
        </dependency>    
```   
(but leave DependencyManagement)
I get a lot of "does not exist" errors for imports of javax.ejb, javax.annotation, etc

**Comment by lprimak** [comment]
Hi,

This is really a maven question, completely out of scope of flowlogix library.
I really don't have free time to spend on support.
If you need more help than the documentation can provide (the library is now very, very well documented)
then you need to talk to corporate about setting up a paid relationship.
FlowLogix does have a corporate GitHub sponsors account, which needs to be done at a corporate level.

**Reply by heatherita** [reply]
I thought this discussion would serve as documentation for future users. But thanks, I'll keep the paid option in mind for future.

**Comment by heatherita** [comment, marked as answer]
Turns out these errors occurred because you had given me an incorrect url for the repo. 
You gave me `https://search.maven.org/`
The correct url was `https://repo1.maven.org/maven2/`

Once I fixed this maven was able to download the pom files correctly.

**Comment by lprimak** [comment]
Full docs are now available at https://docs.flowlogix.com/
